ls1intum/Artemis
# Enhancement/better c template

### Checklist

### Motivation and Context


The course GBS created some new C templates which would be perfect to also be used as our general C template. 

### Description

I copied the C template exercise from the GBS course into our C template.

I also moved the shell scripts in the build jobs into their own template shell file in the resources for every programming language that relies on such a script (C and Python). You now only have to call `readScriptTasksFromTemplate` in order to read all scripts from the template file (also depending on whether sequential runs are enabled or not). This makes maintaining those scripts ways easier while also improving the readability. Especially for scripts that are longer than two lines it doesn't really make sense to have them inline in our Java code as a hardcoded String.

We also always ran the behavior tests for sequential tests for Python and C exercises, even if the structural tests failed. Now, we only run the behavior tests if all structural tests succeeded (as it is intended)

### Steps for Testing


1. Log in to Artemis
2. Navigate to Course Administration
3. Create a new C programming exercise
4. All tests for the solution build should be successful
